megamek/src/megamek/utils/RATGeneratorEditor.java

```diff
@@ -849,41 +853,54 @@ public void setValueAt(Object value, int row, int col) {
             } else if (!((String)value).matches("\\d+[\\+\\-]?(:\\d+)?")) {
                 return;
             } else {
-                ar = new AvailabilityRating(unit, era,
+                ar = new AvailabilityRating(getUnitKey(), era,
                         factions.get(row) + ":" + (String)value);            
             }
             data.get(factions.get(row)).set(col - 1, (String)value);
-            if (rg.getChassisRecord(unit) != null) {
+            if (rg.getChassisRecord(getUnitKey()) != null) {
                 if (ar == null) {
-                    rg.removeChassisFactionRating(era, unit, factions.get(row));                
+                    rg.removeChassisFactionRating(era, getUnitKey(), factions.get(row));                
                 } else {
-                    rg.setChassisFactionRating(era, unit, ar);
+                    rg.setChassisFactionRating(era, getUnitKey(), ar);
                 }
             } else {
                 if (ar == null) {
-                    rg.removeModelFactionRating(era, unit, factions.get(row));               
+                    rg.removeModelFactionRating(era, getUnitKey(), factions.get(row));               
                 } else {
-                    rg.setModelFactionRating(era, unit, ar);
+                    rg.setModelFactionRating(era, unitRecord.getChassisKey(), unitRecord.getKey(), ar);
                 }
             }
         }
 
         public void addEntry(String faction) {
+            if(unitRecord == null) {
+                return;
+            }
+            
             factions.add(faction);
             ArrayList<String> list = new ArrayList<String>();
             while (list.size() < ERAS.length) {
                 list.add("");
             }
             data.put(faction, list);
+            for (int i = 0; i < ERAS.length; i++) {
+                AvailabilityRating ar = new AvailabilityRating(getUnitKey(), ERAS[i],
+                            faction + ":" + 0);
+                if (mode == MODE_MODEL) {
+                    rg.setModelFactionRating(ERAS[i], unitRecord.getChassisKey(), unitRecord.getKey(), ar);
+                } else {
+                    rg.setChassisFactionRating(ERAS[i], getUnitKey(), ar);
+                }
+            }
```

Comment:
Zero and blank have different meanings. Assigning an explicit zero blocks inheritance. It's used for situations where a unit is used by, for example, all IS factions except DC. Or all DC commands except the Sword of Light.

Reply:
What should the default availability rating for a new row in that table be then?

Reply:
Nothing

Reply:
Pretty sure that winds up reading as 0 anyway in practice, since it's stored as an int in the AvailabilityRating data structure.

Reply:
You shouldn't be adding AvailablityRatings at all. This block shouldn't be here

Reply:
Ok. I checked it without adding the availability ratings and it still works fine. I re-worked the validation since we're not actually guaranteed to have a chassis record for any particular era, so it just looks to see if there's any chassis record defined, period.